index.js

```diff
@@ -38,6 +40,28 @@ function BroccoliMergeTrees(inputNodes, options) {
   this._currentTree = FSTree.fromPaths([]);
 }
 
+function joinNodes(host, options, inputNodes) {
+  var nodes = []
+  for (var i = 0 ; i < inputNodes.length ; i++ ) {
+    var node = inputNodes[i];
+    if (host[ASSIMILATION_SYMBOL](options, node)) {
+      for (var j = 0; j < inputNodes[i]._inputNodes.length; j++) {
+        nodes.push(inputNodes[i]._inputNodes[j]);
+      }
+    }
+    else {
+      nodes.push(inputNodes[i]);
+    }
+  }
+
+  return nodes;
+}
+
+BroccoliMergeTrees.prototype[ASSIMILATION_SYMBOL] = function(options, other) {
+  return typeof other[ASSIMILATION_SYMBOL] === 'function' &&
```

Comment:
Can we store the result here, and add a debug call to say if we are or are not assimilating?